core/app/models/concerns/spree/display_money.rb

```diff
@@ -27,7 +27,9 @@ def money_methods(*args)
           define_method("display_#{method_name}") do |**args_list|
             default_opts = respond_to?(:currency) ? { currency: currency } : {}
 
-            Spree::Money.new(send(method_name, **args_list), default_opts.merge(opts).merge(args_list.slice(:currency)))
+            amount = args_list.blank? ? send(method_name) : send(method_name, **args_list)
+
+            Spree::Money.new(amount, default_opts.merge(opts).merge(args_list.slice(:currency)))
```

Comment:
Tried earlier:
`Spree::Money.new(send(*[method_name, **args_list].reject(&:blank?)), default_opts.merge(opts).merge(args_list.slice(:currency)))` but did not work for ruby 3.*